goonj/rule_engine.py

```diff
@@ -0,0 +1,31 @@
+
+class RuleEngine(object):
+
+
+
+
+    def __init__(self,error_code_config):
+        self.error_code_config=error_code_config
+        self.error_count={}
+
+    def is_alerting_required(self,error_code):
+
+
+        if error_code in self.error_code_config.error_code_config:
```

Comment:
Please rename self.error_code_config to something better. `self.error_code_config.error_code_config` is confusing. Someone might mistakenly pass `self.error_code_config.error_code_config` when they need to pass `self.error_code_config` or vice versa